feat: n-menu add renderLabel prop (#191)
TuSimple/naive-ui: src/menu/demos/enUS/render-label.demo.md

# Render Label

The `renderLabel` can be used to batch process the `label` we pass in in `options`.

```html
<n-menu :options="menuOptions" :render-label="renderMenuLabel" />
```

```js
import { h } from 'vue'
import { RouterLink } from 'vue-router'

const menuOptions = [
  {
    label: 'Home',
    key: '/'
  },
  {
    label: 'Introduction',
    key: '/en-US/os-theme/docs/introduction'
  },
  {
    label: 'Components',
    key: '/en-US/os-theme/components'
  }
]

export default {
  setup () {
    return {
      menuOptions,
      renderMenuLabel (option) {

>         if (typeof option.label === 'function') {
>           return option.label()
>         }

Reviewer: ```suggestion
```

Author: 我觉得demo里有必要提醒用户这里有 function 的情况

Reviewer: 但是 Demo 本身没有，用户应该是能意识到的，不然一渲染的话你在 menu 上看到 `Function xxx () { ... }` 肯定就意识到问题了

# Render Label

> The `renderLabel` can be used to batch process the `label` we pass in in `options`.

Reviewer: ```suggestion
The `renderLabel` can be used to batch process the `label` rendering.
```

```html
<n-menu :options="menuOptions" :render-label="renderMenuLabel" />
```

```js
import { h } from 'vue'
import { RouterLink } from 'vue-router'
import { NIcon } from 'naive-ui'
import {
  HomeOutline as HomeIcon,
  BookOutline as BookIcon,
  DocumentTextOutline as DocumentTextIcon
} from '@vicons/ionicons5'

function renderIcon (icon) {
  return () => h(NIcon, null, { default: () => h(icon) })
}

const menuOptions = [
  {
    label: () =>
      h(
        'a',
        {
          href: 'https://www.naiveui.com/en-US/os-theme',
          target: '_blank',
          rel: 'home'
        },
        'Home'

>       ),

Reviewer: 这个换成一致的例子就行，不用专门写一个 function。
